source/hpack.c

```diff
@@ -552,6 +526,96 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
     return AWS_OP_SUCCESS;
 }
 
+enum aws_hpack_decode_status aws_hpack_decode_integer(
+    struct aws_hpack_context *context,
+    struct aws_byte_cursor *to_decode,
+    uint8_t prefix_size,
+    uint64_t *integer) {
+
+    AWS_PRECONDITION(context);
+    AWS_PRECONDITION(to_decode);
+    AWS_PRECONDITION(prefix_size <= 8);
+    AWS_PRECONDITION(integer);
+
+    /* #TODO once frame decoders go away, but for now this is necessary to avoid asserting in fuzz tests */
+    if (to_decode->len == 0) {
+        aws_raise_error(AWS_ERROR_SHORT_BUFFER);
+        return AWS_OP_ERR;
```

Comment:
raise the proper error so we end up reporting AWS_H2_ERR_COMPRESSION_ERROR

Reply:
This should be the proper error. Any decoder function calling this function will see the error, potentially log it, and then report ERROR_COMPRESSION upwards.